auth_saml/models/auth_saml_provider.py

```diff
@@ -408,21 +408,31 @@ def action_refresh_metadata_from_url(self):
         )
         if not providers:
             return False
+
+        providers_to_update = {}
+        for provider in providers:
+            document = requests.get(provider.idp_metadata_url)
```

Comment:
Maybe add a timeout to the GET call. This will ensure that the login fails within a reasonable amount of time. 

```suggestion
            document = requests.get(provider.idp_metadata_url, timeout=5)
```